src/main/java/org/medicmobile/webapp/mobile/AppNotificationManager.java

```diff
@@ -0,0 +1,186 @@
+package org.medicmobile.webapp.mobile;
+
+import static org.medicmobile.webapp.mobile.MedicLog.log;
+
+import android.Manifest;
+import android.app.Activity;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Build;
+
+import androidx.core.app.ActivityCompat;
+import androidx.core.app.NotificationCompat;
+import androidx.core.content.ContextCompat;
+import androidx.work.ExistingPeriodicWorkPolicy;
+import androidx.work.PeriodicWorkRequest;
+import androidx.work.WorkManager;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.medicmobile.webapp.mobile.util.AppDataStore;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.concurrent.TimeUnit;
+
+public class AppNotificationManager {
+	private static final String CHANNEL_ID = "cht_android_notifications";
+	private static final String CHANNEL_NAME = "CHT Android Notifications";
+	public static final int REQUEST_NOTIFICATION_PERMISSION = 1001;
+	public static final String TASK_NOTIFICATIONS_KEY = "task_notifications";
+	public static final String TASK_NOTIFICATION_DAY_KEY = "cht_task_notification_day";
+	public static final String LATEST_NOTIFICATION_TIMESTAMP_KEY = "cht_task_notification_timestamp";
+
+	private final Context context;
+	private final NotificationManager manager;
+	private final String appUrl;
+	private final AppDataStore appDataStore;
+
+	public AppNotificationManager(Context context, String appUrl) {
```

Comment:
Minor, but I guess we can just unpack the app url fom the given `context` instead of requiring it here as a parameter.  


```java
		SettingsStore settings = SettingsStore.in(context);
		String appUrl = settings.getAppUrl();
```